### ISSUE TITLE
Reconsider 'expanded' naming scheme used in generated subpackages

### ISSUE THREAD

**Issue by slimsag** [gopherjs/vecty#163]
I am thinking that it _could_ be valuable to switch from `elem.Header1` to `elem.H1` The reason I say this is because:

1. Sometimes you know the tag name easily (`a`) but not what it represents (`Anchor`).
2. When using any browser dev tools (chrome inspector, etc), you'll see tag names like `h1` not their expanded form.
3. When styling in CSS, you must use their tag name `h1` not `Header1`.

The downside to this change is that it's yet another massive breaking change, so we shouldn't do it needlessly. That said, Vecty is experimental and pre-1.0 still. We could probably offer something like [gofix](https://blog.golang.org/introducing-gofix) quite easily (IIRC it's quite easy to fork/extend for this purpose in specific).

What do others think? Is `elem.H1` or `elem.Header1` more readable and usable? `elem.A` vs. `elem.Anchor`, etc.?

**Comment by slimsag** [comment]
Note we wouldn't do this until #136 is fixed most likely.

**Comment by slimsag** [comment]
A new user suggests over Slack that:

> I would suggest having `elem.Anchor` then having docs say, "If you're looking for `<a>`, use `elem.Anchor`".

**Comment by slimsag** [comment]
(I'm still torn between `elem.H1` vs `elem.Header1`)

**Comment by shurcooL** [comment]
I'm okay with both, as long as it's consistent.

I find it pretty easy to get used to the long-form names. I might have to look up that it's `Anchor` the first time I use that element from vecty, but not the 2nd, 3rd, ..., 100th time that I use it. Same with heading elements.

Maybe if you gave examples of less commonly used names, that'd help provide motivation to switch to tag names.

Perhaps relevant, `x/net/html/atom` package uses original names:

https://godoc.org/golang.org/x/net/html/atom#Atom

**Comment by slimsag** [comment]
> I'm okay with both, as long as it's consistent.

This was my initial reaction, too, but only until I realized the inconsistency between Vecty <-> CSS. How do you feel about this inconsistency? e.g. Go:

```Go
vecty.Achor(...)
```

CSS:
```css
a {
   background: red;
}
```

> Maybe if you gave examples of less commonly used names, that'd help provide motivation to switch to tag names.

Full list here: https://github.com/gopherjs/vecty/blob/1ba2b77c9d1fbf3503d3ddaa1cfc25dff4516ea3/elem/generate.go#L21-L81

**Comment by shurcooL** [comment]
> Full list

I see, that helps.

I like the full names, but given it's very unlikely you'll be able to take on CSS and other places where the abbreviated names will come up, and fix them all everywhere, it might make sense to stick to the web standards. Consistency is more important.

**Comment by pdf** [comment]
I prefer the long-form names personally, and I wish HTML used the long-form for clarity too - you can see that this has consistently been the trend for elements introduced in later spec revisions, as bandwidth constraints (and ubiquitous compression) no longer lend significance to character-level economy on the wire.

That said, as @shurcooL points out, consistency likely trumps personal preference here, particularly for new users.

If we wanted to allow transition without an immediate API breakage, wouldn't be much effort to wrap the (new) short-name funcs in (existing) long-name funcs, with a console warning of the deprecation and impending removal.

**Comment by shurcooL** [comment]
There is another option to consider. It may not be a good idea, but I'll toss it out for consideration anyway.

There could be two `elem`-type packages (not sure about their location). One would offer the long-form element names like `elem.UnorderedList`, while the other would offer the unmodified element names like `elem.UL`.

However, IMO, that simply pushes the (hard) decision of what style to use onto users, which means we (the library creators) are providing less value to users. Specific picky users could define their own such package if they strongly desire. So I wouldn't recommend this route.

**Comment by shurcooL** [comment]
There is another option to consider. It may not be a good idea, but I'll toss it out for consideration anyway.

There could be two `elem`-type packages (not sure about their location). One would offer the long-form element names like `elem.UnorderedList`, while the other would offer the unmodified element names like `elem.UL`.

However, IMO, that simply pushes the (hard) decision of what style to use onto users, which means we (the library creators) are providing less value to users. Specific picky users could define their own such package if they strongly desire. So I wouldn't recommend this route.

**Comment by tbruyelle** [comment]
I share the same opinion, I prefer the long-names, they're more meaningfull.

**Comment by slimsag** [comment]
> However, IMO, that simply pushes the (hard) decision of what style to use onto users, which means we (the library creators) are providing less value to users. Specific picky users could define their own such package if they strongly desire. So I wouldn't recommend this route.

@shurcooL I agree with what you've said here, I don't think that offering both would be ideal in most regards so I'd like to avoid that regardless of which outcome we choose here.

Given what I've heard here from @shurcooL, @pdf, @tbruyelle, and 2(?) newer users over Slack I have consistently heard that using the long-form names is preferred but an easier way to find a mapping of HTML tag names to their Vecty equivalent would be helpful, e.g. the ability to ctrl+f in the godocs for `<a>` and find `Anchor` easily.

I'll conclude that using long-form is the right decision here and change the title of this issue accordingly :)
